tools/gstf_preparation/gstf_preparation.py

```diff
@@ -122,27 +126,33 @@ def feature_to_dict(cols, parent_dict=None):
 def add_gene_to_dict(cols, species, gene_dict):
     global gene_count
     gene = feature_to_dict(cols)
+    if not gene['id']:
+        raise Exception("Id not found among column 9 attribute tags: %s" % cols[8])
+    if 'confidence' in gene and gene['confidence'] != 'high':
```

Comment:
Is `confidence` an attribute found in Ensembl GFF3 files? Never saw it before (same for `representative` for transcripts).